### PR TITLE
docs(kafka source): autogen docs

### LINKED ISSUE #15842: Convert the `kafka` source to use auto-generated docs

### PR DESCRIPTION
**Pull request by davidhuie-dd**
Closes #15842

### CODE REVIEW

**Review by spencergilbert — APPROVED** [review]
LGTM (double checked none of the numbers got changed this time 😉 )

Noting that the example for the librdkafka options isn't rendering on the site, but I think that's an issue on the site side and shouldn't block this - https://deploy-preview-15931--vector-project.netlify.app/docs/reference/configuration/sources/kafka/#librdkafka_options

**Comment by spencergilbert** [comment]
Thinking about this, it's probably worth a new issue to track

File: src/sources/kafka.rs [hunk 1]
```diff
@@ -246,7 +246,7 @@ fn default_headers_key() -> OptionalValuePath {
     OptionalValuePath::from(owned_value_path!("headers"))
 }
 
-fn example_auto_offset_reset_values() -> [&'static str; 7] {
+const fn example_auto_offset_reset_values() -> [&'static str; 7] {
```
Comment (spencergilbert) [inline src/sources/kafka.rs:249]: Was this a clippy suggestion?

Comment (davidhuie-dd) [reply src/sources/kafka.rs:249]: Yep, exactly.

Comment (spencergilbert) [reply src/sources/kafka.rs:249]: 👍 I figured, but also thought I remembered seeing one of these merged that wasn't `const`... probably a different return type 🤷

**Comment by davidhuie-dd** [comment]
@spencergilbert #15957 tracks the issues with HashMap examples.

**Pull request merged**